src/lib.rs

```diff
@@ -260,6 +260,12 @@ pub use os::OsRng;
 
 pub use isaac::{IsaacRng, Isaac64Rng};
 pub use chacha::ChaChaRng;
+pub use sample::{
+    // TODO: `sample` name will be deprecated in 1.0, use `sample_reservoir` instead
```

Comment:
Try `#[depercated(since="VER", note="renamed to sample_reservoir")`. I don't think there's any point *delaying* the deprecation. RFC: https://github.com/rust-lang/rfcs/blob/master/text/1270-deprecation.md